### PR TITLE
feat(ui): Setup browserslist, reduce core-js use

### PR DESCRIPTION
**Pull request by scttcper**
will need to open a PR for getsentry but starting in sentry for discussion

the default we're currently using. browsers like baidu, op_mini, kaios are telling babel preset to bring in all kinds of core-js functions we don't need.

`npx browserslist --env="production"`

```
and_chr 89
and_ff 86
and_qq 10.4
and_uc 12.12
android 89
baidu 7.12
chrome 89
chrome 88
chrome 87
edge 89
edge 88
firefox 86
firefox 85
firefox 78
ie 11
ios_saf 14.0-14.5
ios_saf 13.4-13.7
kaios 2.5
op_mini all
op_mob 62
opera 73
opera 72
safari 14
safari 13.1
samsung 13.0
samsung 12.0
```

this pr
```
chrome 91
chrome 90
chrome 89
chrome 88
chrome 87
chrome 86
chrome 85
chrome 84
chrome 83
chrome 81
edge 91
edge 90
firefox 90
firefox 89
firefox 88
firefox 87
firefox 86
firefox 85
firefox 84
firefox 83
firefox 82
firefox 81
firefox 78
ios_saf 14.5-14.7
ios_saf 14.0-14.4
ios_saf 13.4-13.7
ios_saf 13.3
ios_saf 13.2
ios_saf 13.0-13.1
ios_saf 12.2-12.4
ios_saf 12.0-12.1
op_mob 62
safari 14.1
safari 14
safari 13.1
safari 13
```

from looking at what bundle analyzer says, this change would take us from 215 KB to 91 KB gzipped of core-js

### CODE REVIEW

**Comment by billyvg** [comment]
What % of the global browser population does this list represent?

**Comment by markstory** [comment]
We have a small amount of safari usage for < 13. We don't have any recorded transactions from Baidu, or Kaios. [This discover query has some details](https://sentry.io/organizations/sentry/discover/results/?field=browser.name&field=count%28%29&field=count_unique%28user%29&id=6774&name=Browser+usage&query=event.type%3Atransaction&sort=-count&statsPeriod=7d).

**Comment by scttcper** [comment]
@markstory added last 3 versions of ios but since its updated randomly we'd lose v12 once the next version of ios comes out in a few months. Should we do something like `ios > 12.0` instead?

**Comment by scttcper** [comment]
for some reason the browserslist change is making webpack choke on a circular import we have w/ `StyledSettingsSearch` https://github.com/getsentry/sentry/blob/8fdffed3dc2774b1748e748e73e7ceb97e43cdc3/static/app/views/settings/components/settingsSearch/index.tsx#L57

Will have to resolve this first

**Comment by markstory** [comment]
> Should we do something like ios > 12.0 instead?

Using the ios version is probably better. ios 12 goes back to the 5s which should cover long life users.

File: package.json [hunk 1]
```diff
@@ -230,10 +230,20 @@
   },
   "browserslist": {
     "production": [
-      "defaults"
+      "last 2 Chrome versions",
+      "last 2 Firefox versions",
```
Comment (markstory) [inline package.json:234]: With firefox and chrome on 6 week release cycles we might want to make this longer. What do you think about 10 versions back? That would give us 60 weeks which is just over 1 year.

File: package.json [hunk 2]
```diff
@@ -230,10 +230,20 @@
...
+      "Firefox ESR"
     ],
     "development": [
-      "defaults"
+      "last 2 Chrome versions",
+      "last 2 Firefox versions",
+      "last 2 Edge major versions",
+      "last 2 Safari major versions",
+      "last 3 iOS major versions",
+      "Firefox ESR"
```
Comment (markstory) [inline package.json:246]: I think these constraints are good for development.

Comment (scttcper) [reply package.json:246]: i think i wanted to use the same values as production, i know some people do not.

File: package.json (hunk 1 above)
Comment (scttcper) [reply package.json:234]: yeah we can do that, i'm sure it still has more features than the last 3 major ios versions.

Comment (scttcper) [reply package.json:234]: updated to last 10 versions

**Review by markstory — COMMENTED** [review]
Do we know what the compatibility these shims will have with `Opera Mobile 64`? That browser still gets a decent amount of traffic and likely has less core JS support than last 10 firefox/chrome does.

**Review by markstory — APPROVED** [review]

**Comment by scttcper** [comment]
this is blocked until the circular import is fixed

> for some reason the browserslist change is making webpack choke on a circular import we have w/ `StyledSettingsSearch`
> 
> https://github.com/getsentry/sentry/blob/8fdffed3dc2774b1748e748e73e7ceb97e43cdc3/static/app/views/settings/components/settingsSearch/index.tsx#L57
> 
> Will have to resolve this first

**Comment by scttcper** [comment]
wooooo unblocked, will try to land tomorrow morning

**Review by billyvg — APPROVED** [review]
let's do it

**Pull request merged**